scripts/core/memory.yml

```diff
@@ -440,6 +440,34 @@ returns:
     - $X_RESULT_ERROR_UNSUPPORTED_FEATURE:
         - "If the adapter has no underlying equivalent handle."
 --- #--------------------------------------------------------------------------
+type: function
+desc: "Return platform native mem handle."
+class: $xMem
+name: GetNativeHandleExp
+decl: static
+ordinal: "0"
+details:
+    - "Retrieved native handle can be used for direct interaction with the native platform driver."
+    - "Use interoperability platform extensions to convert native handle to native type."
+    - "The application may call this function from simultaneous threads for the same context."
+    - "The implementation of this function should be thread-safe."
+params:
+    - type: $x_mem_handle_t
+      name: hMem
+      desc: |
+            [in] handle of the mem.
+    - type: $x_device_handle_t
+      name: hDevice
+      desc: |
+            [in] handle of the device that native handle will be resident on.
```

Comment:
I think we should just modify the existing `GetNativeHandle` to add this argument.

Reply:
Is that not an ABI breaking change that will take some time to get merged?

Reply:
We don't claim ABI stability in UR at this time, this won't be an issue as long as the necessary SYCL RT changes are made in tandem with this.